presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestReorderJoins.java

```diff
@@ -61,6 +62,22 @@ public void testEliminateSimpleCrossJoin()
                                 anyTree(tableScan("orders")))));
     }
 
+    @Test
+    public void testEliminateCrossJoinFromTwoJoinBlocks()
+    {
+        String joinToReorder = "(SELECT l.suppkey suppkey FROM part p, orders o, lineitem l WHERE p.partkey = l.partkey AND l.orderkey = o.orderkey)";
```

Comment:
Does it really test the usecase when there are two separate connected join subgraphs?
It looks like one two me.